### PR TITLE
procfs: add cpuinfo

### PR DESCRIPTION
**Pull request by leo11261**
## Summary
add proc/cpuinfo

## Impact
new function

## Testing
custom board. and riscv cpuinfo can be test with [esp32](esp32c3-devkit:nsh)

### CODE REVIEW

**Comment by davids5** [comment]
Where is FS_PROCFS_EXCLUDE_CPUINFO enabled?

**Comment by leo11261** [comment]
> Where is FS_PROCFS_EXCLUDE_CPUINFO enabled?

it defaults to be disabled and can be enable by menuconfig.
this mocro is defined in fs/procfs/Kconfig.

File: arch/risc-v/src/common/Make.defs [hunk 1]
```diff
@@ -23,18 +23,18 @@ STARTUP_OBJS = crt0$(OBJEXT)
...
 # Specify C code within the common directory to be included
-CMN_CSRCS += riscv_initialize.c riscv_swint.c riscv_mtimer.c
-CMN_CSRCS += riscv_allocateheap.c riscv_createstack.c
-CMN_CSRCS += riscv_cpuidlestack.c riscv_doirq.c riscv_exit.c riscv_exception.c
-CMN_CSRCS += riscv_getnewintctx.c riscv_getintstack.c riscv_initialstate.c
-CMN_CSRCS += riscv_idle.c riscv_modifyreg32.c riscv_nputs.c riscv_releasestack.c
-CMN_CSRCS += riscv_registerdump.c riscv_stackframe.c riscv_schedulesigaction.c
-CMN_CSRCS += riscv_sigdeliver.c riscv_switchcontext.c
-CMN_CSRCS += riscv_usestack.c riscv_tcbinfo.c
+CMN_CSRCS += riscv_allocateheap.c riscv_copystate.c riscv_cpuidlestack.c
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/Make.defs:30]: remove riscv_copystate.c

**Review by jerpelea — APPROVED** [review]
please add commit messages explaining the change

**Pull request merged**